pkg/skaffold/runner/new_test.go

```diff
@@ -39,6 +39,7 @@ func TestGetDeployer(tOuter *testing.T) {
 			cfg         latest_v1.DeployType
 			helmVersion string
 			expected    deploy.Deployer
+			apply       bool
```

Comment:
So we're using `apply = true` to implicitly test `getDefaultDeployer()`.  I think it would be better to test `getDefaultDeployer()` explicitly, and add some more cases like Kustomize flags and conflicts.

Reply:
there is actually another set of test cases below this one that tests `getDefaultDeployer()` explicitly - the main thing I changed here is to embed the `if runCtx.Opts.Apply` check in the call to `getDeployer()` which sends us through to `getDefaultDeployer()`, which I thought cleaned up the code in `NewForConfig()` a bit.

the addition of the `apply` boolean to this set of tests just tests the branching logic that ensures if we're in `skaffold apply`, we get a default deployer - the other tests make sure that all the other cases (e.g. Kustomize flags, unresolvable config, etc.) are working as intended.